Generate.py

```diff
@@ -481,7 +432,10 @@ def handle_option(ret: argparse.Namespace, game_weights: dict, option_key: str,
             raise Exception(f"Error generating option {option_key} in {ret.game}") from e
         else:
             if hasattr(player_option, "verify"):
-                player_option.verify(AutoWorldRegister.world_types[ret.game])
+                if player_option.requires_plando:
+                    player_option.verify(plando_options)
+                else:
+                    player_option.verify(AutoWorldRegister.world_types[ret.game])
```

Comment:
What would be more logical is that overwriting verify() is all a class has to do to get custom verify behaviour.
So I would propose to change the signature of verify to always be
`def verify(world: World, plando_options: Generate.PlandoSettings)`
and always pass both arguments.